c7n/resources/cloudtrail.py

```diff
@@ -55,10 +54,8 @@ class IsShadow(Filter):
     embedded = False
 
     def process(self, resources, event=None):
-        anded = lambda x: True # NOQA
-        op = self.data.get('state', True) and anded or operator.__not__
         rcount = len(resources)
-        trails = [t for t in resources if op(self.is_shadow(t))]
+        trails = [t for t in resources if (self.is_shadow(t) == self.data.get('state', True))]
```

Comment:
looks good, not sure what the op stuff was for, what your doing here is much simpler, thanks.